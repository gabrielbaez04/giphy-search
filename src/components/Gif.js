import React from 'react';
import PropTypes from 'prop-types';

const Gif = ({ gif, onGifSelect }) => {
  return (
    <div className="gif-item" onClick={() => onGifSelect(gif)}>
      <img src={gif.images.downsized_still.url} />
      <div className="gif-title">{gif.title.replace("GIF", "")}</div>
    </div>
  )
};
Gif.propTypes = {
  gif: PropTypes.object.isRequired,
  onGifSelect: PropTypes.func.isRequired
}
export default Gif;