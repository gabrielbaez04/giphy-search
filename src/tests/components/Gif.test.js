import { shallow } from 'enzyme';
import Gif from '../../components/Gif';
import React from 'react';

test('should render Gif correctly', () => {
    const gif = { title: "test", images: { downsized_still: { url: "test.url.com" } } }
    const wrapper = shallow(<Gif gif={gif} onGifSelect={() => { }} />);
    expect(wrapper).toMatchSnapshot();
});