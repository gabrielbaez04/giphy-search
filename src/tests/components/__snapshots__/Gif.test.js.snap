// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Gif correctly 1`] = `
<div
  className="gif-item"
  onClick={[Function]}
>
  <img
    src="test.url.com"
  />
  <div
    className="gif-title"
  >
    test
  </div>
</div>
`;
